### ISSUE TITLE
Redeserializing the output of deserialize<object>

### ISSUE THREAD

**Issue by danfrias** [neuecc/Utf8Json#60]
If I use the command:

`JsonSerializer.Deserialize<object> (json)`

to deserialize Json into an object, it creates an object that, effectively, is of type `Dictionary<string, object>`.  My question: is it then possible to cast this object, or part of this object, into a final type, or do I need to deserialize once again from the original JSON to get the data into a specific type?

Thank you.

**Comment by neuecc** [comment]
Deserialize<object> guranteeds deserialize to `IDictionary<string, object>` so you can cast to `IDictionary<string, object>` and build to finaly type manually.

**Comment by velddev** [comment]
For this issue, I was thinking of adding some sort of raw data struct that would keep the object in a byte[] to re-deserialize for later use. I'll see if I can do anything related to this idea.

**Comment by danfrias** [comment]
> For this issue, I was thinking of adding some sort of raw data struct that would keep the object in a byte[] to re-deserialize for later use. I'll see if I can do anything related to this idea.

From a user perspective this would be fantastic and be exactly what I was looking for when I posted the original question.

**Comment by velddev** [comment]
Yeah, same here. I've been working on this, and have a sort of prototype. will push it to my fork soon, and hopefully PR it in later
